entrypoint.sh

```diff
@@ -2,6 +2,32 @@
 
 set -eu
 
+function get_diff() {
+  base="$1"
+  sha="$2"
+  filter="$3"
+  mapfile -t submodules < <(git submodule | awk '{print $2}')
+  for sub in "${submodules[@]}"; do
+    mapfile -t sub_commits < <(
+      git diff "$base" "$sha" -- "$sub" \
+        | grep '^[-+]Subproject commit' \
+        | sed 's/[-+]Subproject commit //'
+      )
+    # this arcane stuff here because of bash<4.4.x idiosyncrasy
+    if [ -n "${sub_commits[0]+"${sub_commits[0]}"}" ]; then
```

Comment:
Based on https://github.com/tj-actions/changed-files/runs/5290699593?check_suite_focus=true#step:6:91

Line 20: Seems to be generating unbound variable errors

```bash
  D:\a\changed-files\changed-files\.//entrypoint.sh: line 20: sub_commits[1]: unbound variable
```

Reply:
Yup, here it doesn't understand bash array syntax. Probably old bash

Reply:
Removed arrays